src/providers/EtherspotContextProvider.tsx

```diff
@@ -378,6 +379,38 @@ const EtherspotContextProvider = ({
     [sdk, accountAddress]
   );
 
+  const getRatesByTokenAddresses = useCallback(async (chainId: number, tokenAddresses: string[]) => {
+    let tokens  = tokenAddresses.filter((address) => address !== null)
+    if(!sdk || !tokens.length) return null
+    try {
+      const rates: RateData = await sdk.fetchExchangeRates({tokens, chainId});
+      if(!rates.errored && rates.items.length > 0) {
```

Comment:
Maybe just add a few null checks considering we're interacting with the sdk and don't know if something will be returned
```suggestion
      if (!rates?.errored && rates?.items?.length) {
```

Reply:
@Yash-Chaubal so we get this key everytime the requests succeed so rates.errored is always there if you see the Type `RateData`